crates/wasmparser/src/binary_reader.rs

```diff
@@ -277,6 +277,18 @@ impl<'a> BinaryReader<'a> {
         })
     }
 
+    fn read_ordering(&mut self) -> Result<Ordering> {
+        let byte = self.read_u8()?;
```

Comment:
Unsure how much in flux this is, but [cross referencing this](https://github.com/WebAssembly/shared-everything-threads/blob/main/proposals/shared-everything-threads/Overview.md#instructions) this should technically be `read_var_u32` as specified currently.

That being said I also think that the u32 is probably overkill, so the spec may also want to switch to `u8` perhaps.

Reply:
Good catch!

Reply:
I can still push a `u8` during encoding, right?